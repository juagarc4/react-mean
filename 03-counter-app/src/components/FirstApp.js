// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, subtitle }) => {


    // const greeting = 'Hola mundo';
    // const user = {
    //     name: 'Raul',
    //     age: 22
    // }
    //<> === <Fragment ><Fragment />
    console.log();
    return (
        <>
            <h1>{ greeting }</h1>
            {/*<pre>{ JSON.stringify(user, null, 3) }</pre>*/}
            <p>{ subtitle }</p>
        </>
    );
}

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired,
}
FirstApp.defaultProps = {
    subtitle : 'Soy un subtitulo',
}
export default FirstApp;