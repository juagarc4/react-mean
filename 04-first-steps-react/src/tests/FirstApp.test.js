import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
// import { render } from '@testing-library/react'
import FirstApp from '../components/FirstApp'

describe('Test <FirstApp/>', () => {
  // test('should show message "Hello World"', () => {
  //   const greeting = 'Hello World'
  //   const { getByText } = render(<FirstApp greeting={greeting} />)

  //   expect(getByText(greeting)).toBeInTheDocument()
  // })
  test('should show component <FirstApp /> right', () => {
    const greeting = 'Hello World'
    const wrapper = shallow(<FirstApp greeting={greeting} />)

    expect(wrapper).toMatchSnapshot()
  })
  test('should show subtitle sent by props', () => {
    const greeting = 'Hello World'
    const subtitle = 'I am a subtitle'
    const wrapper = shallow(<FirstApp greeting={greeting} subtitle={subtitle} />)

    const paragraphText = wrapper.find('p').text()

    expect(paragraphText).toBe(subtitle)
  })
})
