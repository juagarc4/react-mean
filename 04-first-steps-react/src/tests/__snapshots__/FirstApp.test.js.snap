// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp/> should show component <FirstApp /> right 1`] = `
<Fragment>
  <h1>
    Hello World
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
